CHANGELOG.md

```diff
@@ -4,6 +4,162 @@ All notable changes to Router will be documented in this file.
 
 This project adheres to [Semantic Versioning v2.0.0](https://semver.org/spec/v2.0.0.html).
 
+# [1.30.0] - 2023-09-14
+
+## 🚀 Features
+
+### Rhai Support at the `router_service` ([Issue #2278](https://github.com/apollographql/router/issues/2278))
+
+It is now possible to interact with requests and responses at the `router_service` level [using Rhai-based customizations](https://www.apollographql.com/docs/router/customizations/rhai/). The functionality is very similar to that provided for interacting with existing services, for example `supergraph_service`. For instance, you may "map" requests and responses as follows:
+
+```rust
+fn router_service(service) {
+    const request_callback = Fn("process_request");
+    service.map_request(request_callback);
+    const response_callback = Fn("process_response");
+    service.map_response(response_callback);
+}
+```
+
+The main difference from [existing services](https://www.apollographql.com/docs/router/customizations/rhai/#router-request-lifecycle) is that the `router_service` allows operating on _HTTP bodies_ rather than the more structured representations available at later service layers, like the [supergraph service](https://www.apollographql.com/docs/router/customizations/rhai/#supergraphservice).
+
+This means that `Request.body` and `Response.body` are both strings, which provides flexibility to make more substantial modifications to the body.  Users should choose the most convenient service for the types of changes they're needing to make, since operating on the strings will require more work than operating on properties of the structured objects in the supergraph service.
+
+As another example, this logs the bodies of requests:
+
+```rust
+// Generate a log for each request at this stage
+fn process_request(request) {
+    print(`body: ${request.body}`);
+}
+
+// Generate a log for each response at this stage
+fn process_response(response) {
+    print(`body: ${response.body}`);
+}
+```
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3234
+
+## 🐛 Fixes
+
+### Small performance improvements to telemetry ([PR #3656](https://github.com/apollographql/router/pull/3656))
+
+We applied some small performance improvements to the `SpanMetricsExporter` (which is used to report span timings), some of which apply in cases where telemetry is disabled and could be apparent to most users.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3656
+
+### Handle interfaces in fragment spreads when `__typename` is omitted ([Issue #2587](https://github.com/apollographql/router/issues/2587))
+
+We now check the parent type when using an inline-fragment, rather than relying on the expectation that `__typename` will be present, since that is not a guarantee.  For cases where `__typename` was being omitted, this corrects cases where a portion of the selection set was silently dropped and not returned.
```

Comment:
```suggestion
Parent type when using an inline-fragment is verified, rather than relying on the expectation that `__typename` will be present.  For cases where `__typename` was being omitted, this corrects responses where a portion of the selection set was silently dropped and not returned.
```